### ISSUE TITLE
Recoill 0.0.11 - window is not defined - SSR

### ISSUE THREAD

**Issue by lednhatkhanh** [facebookexperimental/Recoil#578]
```
nextjs: 9.5.3
recoil: 0.0.11
```

```
ReferenceError: window is not defined
    at Object.<anonymous> (/Users/khanh/workspace/personal/next-thullo/node_modules/recoil/cjs/recoil.js:1985:3)
    at Module._compile (internal/modules/cjs/loader.js:1138:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10)
    at Module.load (internal/modules/cjs/loader.js:986:32)
    at Function.Module._load (internal/modules/cjs/loader.js:879:14)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at eval (webpack-internal:///recoil:1:18)
    at Object.recoil (/Users/khanh/workspace/personal/next-thullo/.next/server/pages/_app.js:182:1)
    at __webpack_require__ (/Users/khanh/workspace/personal/next-thullo/.next/server/pages/_app.js:23:31)
```

**Comment by lednhatkhanh** [comment]
Downgrade `recoil` to version `0.0.10` fixed the issue for me

**Comment by davidmccabe** [comment]
Fixed in 0.0.12. 

I’d like to figure out how to either make window undefined in jest tests, make it not a global for flow, or preferably both. Couldn’t find how to do either yet. Anyone know?
